pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -167,7 +192,9 @@ def handle_thinking_delta(
         signature: str | None = None,
         provider_name: str | None = None,
         provider_details: dict[str, Any] | None = None,
-    ) -> ModelResponseStreamEvent:
+        raw_content_delta: str | None = None,
+        raw_content_index: int = 0,
+    ) -> Iterator[ModelResponseStreamEvent]:
```

Comment:
I got confused writing the callback so I ended up with this instead. lmk what you think